### PR TITLE
Move package future from extended to core

### PR DESCRIPTION
**Pull request by mandeepsplaha**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
This PR moves package `future` from the extended packages to core packages. This package is a requirement for `pytorch` golden container (pytorch spec will be added to Mariner soon).

###### Change Log  

- Package `future` is moved from SPECS-EXTENDED to SPECS, which results in the package becoming a core package.

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

Tested locally as well as in buddy build.
- **AMD64** Buddy Build Pipeline Build Id: [296930](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=296930&view=results)
- **ARM64** Buddy Build Pipeline Build Id: [296931](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=296931&view=results)

### CODE REVIEW

**Review by rikenm1 — CHANGES_REQUESTED** [review]
Can you run spec-linter? The formatting is weird here. 
Also can you remove all the python2 stuff? Usually, when we are bringing packages from the Extended to Core, we clean a lot of If conditions if they do not apply to CBL-Mariner.  Python2 is one of them. We do not have python2.

**Comment by oliviacrain** [comment]
Quick question first- are we building PyTorch with caffe2 support? Just wondering if we truly need this dependency. A [PR was merged to the PyTorch upstream a few days ago](https://github.com/pytorch/pytorch/pull/92596) to remove this dependency from the caffe2 codebase. It looks like this dependency is only in the caffe2 codebase. For reference, caffe2 is a deprecated framework that isn't build by default as part of PyTorch, but lives in the same source tree.

File: SPECS/future/future.spec [hunk 1]
```diff
@@ -38,42 +33,41 @@ BuildRequires:  python%{python3_other_pkgversion}-devel
 
 %if 0%{?with_python2}
```
Comment (PawelWMS) [inline SPECS/future/future.spec:34]: Let's also remove all bits related to Python 2. It's not coming back.

File: SPECS/future/future.spec [hunk 2]
```diff
@@ -38,42 +33,41 @@ BuildRequires:  python%{python3_other_pkgversion}-devel
...
-Summary: Easy, clean, reliable Python 2/3 compatibility
 %{?python_provide:%python_provide python2-%{name}}
+Summary:        Easy, clean, reliable Python 2/3 compatibility
+BuildRequires:  python2-devel
+BuildRequires:  python2-numpy
+BuildRequires:  python2-pytest
+BuildRequires:  python2-requests
+BuildRequires:  python2-setuptools
+Provides:       future = 0:%{version}-%{release}
 %if 0%{?el6}
```
Comment (PawelWMS) [inline SPECS/future/future.spec:44]: Similarly, let's also remove macros referencing other distros.

File: SPECS/future/future.spec [hunk 3]
```diff
@@ -38,42 +33,41 @@ BuildRequires:  python%{python3_other_pkgversion}-devel
...
-BuildRequires: python2-numpy
-BuildRequires: python2-requests
-BuildRequires: python2-pytest
-Provides:      future = 0:%{version}-%{release}
-
 %description -n python2-%{name}
 %{_description}
 %endif
 
 %if 0%{?with_python3}
```
Comment (PawelWMS) [inline SPECS/future/future.spec:56]: I think we can assume `with_python3` is always on.

**Comment by mandeepsplaha** [comment]
> 

Thanks for bringing the point about caffe2, Olivia. You're right, we don't need the future package for pytorch at runtime. I will be closing this PR.

File: SPECS/future/future.spec (hunk 3 above)
Comment (mandeepsplaha) [reply SPECS/future/future.spec:56]: Closing the PR because this package doesn't need to be moved to core.

**Pull request closed without merging**